docs/architecture/Overview.md

```diff
@@ -0,0 +1,164 @@
+Documentation in this directory dives into the _why_.
+
+## Architecture - layering
+
+At a high level, the library exposes two distinct layers of APIs.
+
+- A lower-abstraction API in the form of UI components, stateful ACS modality clients (chat / calling) and bindings between the two. The focus in this layer is in giving the customers maximum flexibility. The API makes the default use-cases easier but not at the cost of making non-default use-cases impossible.
+
+- A higher-abstraction API in the form of UI composites, powered by adapters. The focus in this layer is in addressing specific end-to-end use-cases well. While this layer does provide customers some customizability, it choses to make some non-default use-cases hard or impossible. By restricting the set of supported use-cases, this layer is able to support the covered use-cases more extensively and with minimal effort from Contoso.
+
+The user-facing documentation in [storybook](https://azure.github.io/communication-ui-library) starts with the composite layer, as it's the easiest API to use for new customers. It then progressively reveals the components API as an alternative for power-users.
+
+This document starts with the component layer, as it forms the foundation for the composite layer in addition to being a public API customers can use.
+
+## Component layer
+
+The component layer provides a set of UI components and bindings that allow the components to work with the ACS backend services. The architecture follows [unidirectional data flow](./ComponentDesign.md).
+
+A typical application that uses the component layer API looks as follows:
+
+
+
+        ┌────────────────────────────────────────────────────────┐
+        │                                                        │
+        │  Contoso Application                                   │
+        │   * Add top-level Providers using StatefulClients      │
+        │   * Use UI components in app, with component bindings  │
+        │    ┌──────────────────────────────────────┐            │
+        │    │  UI Components                       │            │
+        │    │   * Buttons                          │            │
+        │    │   * CallControlBar                   │            │
+        │    │   * VideoGallery                     │            │
+        │    │   * MessageThread                    │            │
+        │    │   * ...                              │            │
+        │    │                                      │            │
+        │    └───┬────────────────────────────▲─────┘            │
+        │        │                            │                  │
+        │        │                            │                  │
+        │    ┌───┼────────────────────────────┼──────┐           │
+        │    │   │   Component Bindings       │      │           │
+        │    │   │                            │      │           │
+        │    │ Handlers                    Selectors │           │
```

Comment:
Would be nice to add one or two example for handlers and selectors like above and below

Handlers
* onToggleCamera()
* ...

Selectors
* getDisplayName()
* ...

The target is to help experienced developer understand the architecture without reading any text below (if possible)